### PR TITLE
Step to goal

### LINKED ISSUE #94: User can review their progress
This view consists of: 
- goal tile (reuse component)
- sliders to show 3 latest ratings (reusable components)
- preview of line chart with button to expand view of line chart
- button to submit a new rating

Things to do:

- [x]  action for selecting goal from list
- [x]  switch view when selecting a goal
- [x]  create and style component for reviewing a goal
- [x]  mock line chart
- [x]  button to create new rating
- [x]  show goal tile
- [x]  slider things to show recent progress
- [x]  slider animations

### PR DESCRIPTION
**Pull request by shiryz**
Fixes #94 .

### CODE REVIEW

File: app/components/view-goal/view-goal.jsx [hunk 1]
```diff
@@ -1,8 +1,48 @@
...
+
+const viewGoal = ( { currentGoal } ) => {
+  const rateGoal = (goal) => {
+    console.log("hello");
+  };
+  const rating = [5, 6, 7];
+  const progressBars = rating.map( rating =>
+     <ProgressBarComponent
+       key={rating}
+       rating={rating}
```
Comment (des-des) [inline app/components/view-goal/view-goal.jsx]: Hmm so we make 3 progress bars, one with rating 6, one with rating 7, one with rating 8.

I find this generally confusing.

File: app/components/view-goal/view-goal.jsx [hunk 2]
```diff
@@ -1,8 +1,48 @@
...
+       time={"today"} />
+   ); {/* TODO: Change the key to id */}
```
Comment (des-des) [inline app/components/view-goal/view-goal.jsx]: todo = code smell

File: app/components/view-goal/view-goal.jsx [hunk 3]
```diff
@@ -1,8 +1,48 @@
...
+        <div className="button-outer">
+          <button
+            type="button"
+            name="button"
+            className="new-rating-button"
+            onClick={ () => rateGoal(currentGoal) }
+          >New Rating</button>
+        </div>
+      </div>
+  </div>
```
Comment (des-des) [inline app/components/view-goal/view-goal.jsx]: indentation

File: app/scss/_view-goal.scss [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+  width: 20px;
+  border-radius: 50%;
+  position: absolute;
+  background-color: $light;
+  border: $blueBorderThin;
+  margin-top: -14px;
+  top: 48%;
+  p {
+    top: -20%;
+    text-align: center;
```
Comment (des-des) [inline app/scss/_view-goal.scss]: Hmm, there are lots of `20px`'s going around. Are some of them the same 20?

File: tests/reducer-test.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
+const tape = require('tape');
```
Comment (des-des) [inline tests/reducer-test.js]: I have already looked at this diff!

File: app/components/view-goal/view-goal.jsx [hunk 6]
```diff
@@ -1,8 +1,48 @@
 import React from 'react';
+import GoalTileComponent from '../goal-tile.jsx'
+import ProgressBarComponent from './progress-bar.jsx'
+
+const viewGoal = ( { currentGoal } ) => {
+  const rateGoal = (goal) => {
+    console.log("hello");
```
Comment (des-des) [inline app/components/view-goal/view-goal.jsx]: Hmm.. It is better to get actions / reducers in place, then to build the view, then to be using fake actions..

Comment (des-des) [inline app/components/view-goal/view-goal.jsx]: Delete this and remove from jsx below as if for debugging and will take 10s to put back.

File: app/components/view-goal/view-goal.jsx (hunk 1 above)
Comment (des-des) [inline app/components/view-goal/view-goal.jsx]: Why does a progress bar have a rating? (its a naming thing)

File: app/components/view-goal/progress-bar.jsx [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+  const left = {
+    left: `${(rating*10)-6}%`,
+    animation: `slideRight${rating} 2s ease-out`};
+  return (
+    <div className="progress-bar-container">
+      <div className="progress-bar-background">
+
+      </div>
+      <div className="progress-bar-forground" style={style}>
+
```
Comment (emilyb7) [inline app/components/view-goal/progress-bar.jsx]: also: change spelling to `foreground`

File: app/components/view-goal/view-goal.jsx (hunk 1 above)
Comment (shiryz) [inline app/components/view-goal/view-goal.jsx]: each day has different progress rating, that's why we have different progress bars

File: app/scss/_view-goal.scss (hunk 4 above)
Comment (shiryz) [inline app/scss/_view-goal.scss]: nope they're unrelated.

File: app/components/view-goal/view-goal.jsx (hunk 1 above)
Comment (emilyb7) [inline app/components/view-goal/view-goal.jsx]: not a progress bar in the typical UI sense. But very much a bar that shows progress!

File: app/components/view-goal/progress-bar.jsx [hunk 8]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+
+const ProgressBarComponent = ({ rating, time }) => {
```
Comment (des-des) [inline app/components/view-goal/progress-bar.jsx]: I think in the context of a progress bar, `progress` would be a better name than `rating`.

File: app/components/view-goal/view-goal.jsx [hunk 9]
```diff
@@ -1,8 +1,44 @@
 import React from 'react';
+import GoalTileComponent from '../goal-tile.jsx'
+import ProgressBarComponent from './progress-bar.jsx'
+
+const viewGoal = ( { currentGoal } ) => {
+  const ratings = [5, 6, 7];
```
Comment (des-des) [inline app/components/view-goal/view-goal.jsx]: Remove mock data. Can move into initial state of reducer

**Comment by des-des** [comment]
Yay

**Pull request merged**